persistence-test/src/main/java/io/stargate/it/PersistenceTest.java

```diff
@@ -1964,6 +1966,42 @@ public void testRowDecorator() throws ExecutionException, InterruptedException {
     assertGt(last, first, dec1, dec2);
   }
 
+  @Test
+  public void testRowDecoratorComparableBytes() throws ExecutionException, InterruptedException {
+    setupCustomPagingData();
+
+    // Obtain partition keys in "ring" order
+    AbstractBound<?> selectAll =
+        dataStore
+            .queryBuilder()
+            .select()
+            .column("pk")
+            .column("val")
+            .from(keyspace, table)
+            .build()
+            .bind();
+
+    ResultSet rs1 = dataStore.execute(selectAll).get();
+    RowDecorator dec1 = rs1.makeRowDecorator();
+    List<Row> rows = rs1.rows();
+    if (backend.isDse()) {
+      List<String> allComparableByteStrings =
+          rows.stream()
+              .map(dec1::getComparableBytes)
+              .map(ByteComparable::fixedLength)
+              .map(x -> x.byteComparableAsString(ByteComparable.Version.DSE68))
+              .collect(Collectors.toList());
+      Collections.sort(allComparableByteStrings);
+      for (int idx = 0; idx < allComparableByteStrings.size(); idx++) {
+        String cb1 = allComparableByteStrings.get(idx);
+        String cb2 =
+            ByteComparable.fixedLength(dec1.getComparableBytes(rows.get(idx)))
+                .byteComparableAsString(ByteComparable.Version.DSE68);
+        assertThat(cb1.compareTo(cb2)).isEqualTo(0);
+      }
+    }
```

Comment:
Not sure what's going on here, why simply not using and adapting the code from the above method: 

```
while (it1.hasNext()) {
      assertThat(it2.hasNext()).isTrue();
      Row r1 = it1.next();
      Row r2 = it2.next();
      first = first == null ? r1 : first;
      last = r1;
      assertEq(r1, r2, dec1, dec2);
      if (p1 == null) {
        p1 = r1;
      }
      assertGtEq(r1, p1, dec1, dec2);
    }
    assertThat(it2.hasNext()).isFalse();
    assertGt(last, first, dec1, dec2);
}    
```